llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -4312,6 +4312,10 @@ static Value *simplifyWithOpReplaced(Value *V, Value *Op, Value *RepOp,
   if (match(I, m_Intrinsic<Intrinsic::is_constant>()))
     return nullptr;
 
+  // Don't simplify a use of a multi-use freeze.
+  if (isa<FreezeInst>(I) && !I->hasOneUse())
```

Comment:
We cannot check uses in InstSimplify. There could be multi-use down-stream and not all of them replaced, or similar. We should just skip freeze entirely.

Reply:
Done.